File: CoinBlock.java
import java.awt.Graphics;
import java.awt.Image;
// import sprites.*;


class CoinBlock extends Sprite {
  static Image[] images = {};
  int imgNum;
  int numCoinsSpit;
  int cooldown;

  CoinBlock (int x, int y) {
    super(x, y, 89, 83);
    loadImages();
  }

  CoinBlock (Json ob) {
    super();
    x = (int) ob.getLong("x");
    y = (int) ob.getLong("y");
    w = (int) ob.getLong("w");
    h = (int) ob.getLong("h");
    loadImages();
  }

  CoinBlock (CoinBlock cb) {
    super(cb);
    imgNum = cb.imgNum;
    numCoinsSpit = cb.numCoinsSpit;
    cooldown = cb.cooldown;
  }

  private void loadImages() {
    if (images.length == 0) {
      images = new Image[2];
      images[0] = View.loadImage("imgs/block1.png");
      images[1] = View.loadImage("imgs/block2.png");
    }
  }

  public Json marshall() {
    Json ob = Json.newObject();
		ob.add("x", x);
		ob.add("y", y);
		ob.add("w", w);
		ob.add("h", h);
		ob.add("type", "coinBlock");
		return ob;
  }

  public void update(Model m) {
    if (cooldown > 0) {
      cooldown--;
    }
  }

  public void draw (Graphics g, Model m) {
    imgNum = (numCoinsSpit >= 5) ? 1 : 0;
    g.drawImage(images[imgNum], x - m.camX, y - m.camY, null);
  }

  @Override
  public void spriteHit(Model m, Sprite s) {
    if (s.isMario && s.y > y+h-30 && numCoinsSpit < 5 && cooldown <= 0) { //if it is mario and he is below me
      Coin c = new Coin(this);
      System.out.println("Hit");
      m.mario.coins++;
      m.addSprite(c);
      cooldown = 12;
      numCoinsSpit++;
    }
  }

}

File: coinblock.py
import pygame
from sprite import Sprite
from coin import Coin

class CoinBlock(Sprite):
  def __init__(self,x, y):
    self.x = x
    self.y = y
    self.w = 89
    self.h = 83
    self.images = []
    self.cooldown = 12
    self.spitCount = 0
    self.loadImages()

  def loadImages(self):
    for i in range(1,3):
      imgStr = "imgs/block" + str(i) + ".png"
      self.images.append(pygame.image.load(imgStr))

  def update(self, model):
    if self.cooldown > 0:
      self.cooldown -= 1

  def draw(self, screen, model):
    imgNum = 1 if self.spitCount >= 5 else 0
    rect = pygame.Rect(self.x - model.camx, self.y, self.w, self.h)
    screen.blit(self.images[imgNum], rect)

  def spriteHit(self, model, sprite):
    if sprite.isMario and sprite.y > self.y + self.h - 2 and self.spitCount < 5 and self.cooldown <= 0:
      c = Coin(self)
      model.sprites.append(c)
      self.cooldown = 12
      self.spitCount += 1
